src/xmipp/libraries/reconstruction/angular_assignment_mag.h

```diff
@@ -0,0 +1,212 @@
+/***************************************************************************
+ *
+ * Authors:     Jeison Méndez García (jmendez@utp.edu.co)
+ *
+ * Instituto de Investigaciones en Matemáticas Aplicadas y en Sistemas -- IIMAS
+ * Universidad Nacional Autónoma de México -UNAM
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+ * 02111-1307  USA
+ *
+ *  All comments concerning this program package may be sent to the
+ *  e-mail address 'xmipp@cnb.csic.es'
+ ***************************************************************************/
+
+#ifndef __ANGULAR_ASSIGNMENT_MAG_H
+#define __ANGULAR_ASSIGNMENT_MAG_H
+
+#include "core/xmipp_metadata_program.h"
+#include "core/matrix1d.h"
+#include "core/multidim_array.h"
+#include "core/metadata_vec.h"
+
+#include <core/xmipp_program.h>
+#include <core/xmipp_fftw.h>
+#include <core/metadata_extension.h>
+#include <core/multidim_array.h>
+#include <core/symmetries.h>
+#include <core/xmipp_image.h>
+//#include <core/metadata.h>
+#include <core/utils/memory_utils.h>
+#include <data/mask.h>
+#include <data/filters.h>
+#include "data/projection.h"
+#include "data/fourier_projection.h"
+#include <reconstruction/project_real_shears.h>
+
+#include <vector>
+#include <fstream> 
+#include <ctime>
+
+
+/**@defgroup AngularAssignmentMag ***
+   @ingroup ReconsLibrary */
+//@{
+
+/** Angular_Assignment_mag parameters. */
+class ProgAngularAssignmentMag: public XmippMetadataProgram
+{
+public:
```

Comment:
use `private` by default. Only those methods / fields that need to be `public` should be `public`

Reply:
Done

Reply:
I just need to have 2 or 3 public variables, but SonarCloud code analysis said is a codeSmell to have a mix with public and private variables.

Reply:
I don't know what you mean. Please send me a link from SonarCloud.
What you want is:
```
{
public:
// list public methods
// list public attributes (ideally none)
private:
// the rest goes here, I prefer to put methods first and variables at the end
}